bindings/cpp/iiopp.h

```diff
@@ -561,9 +562,9 @@ class Device : public impl::IndexedSequence<Device, Channel>
     optstr label() const { return impl::opt(iio_device_get_label(p));}
     unsigned int channels_count() const {return iio_device_get_channels_count(p);}
     unsigned int attrs_count() const {return iio_device_get_attrs_count(p);}
-    Channel channel(unsigned int idx) const {return Channel{iio_device_get_channel(p, idx)};}
+    optional<Channel> channel(unsigned int idx) const {return impl::maybe<Channel>(iio_device_get_channel(p, idx));}
```

Comment:
From my review this is a good solution. However we need to extend the design concept all the way to iiopp::Context. E.g. ```Device iiopp::Context::find_device(cstr name);``` Still suffers from the assertion bug as shown by the example below:
```
#include <iiopp.h>
#include <iostream>

int main(int argc, char ** argv)
{
    try
    {
        auto iio_ctx = iiopp::create_context(nullptr, nullptr);
        auto dev = iio_ctx->find_device("aaxi_fan_control"); // This results in an assert
    }
    catch (iiopp::error & e)
    {
        std::cerr << "ERROR " << e.code().value() << ": " << e.what() << std::endl;
        return EXIT_FAILURE;
    }

    return EXIT_SUCCESS;
}
```
g++ -ggdb3 -I/usr/include/iio -o cpp_test cpp_test.cpp -liio